datalad/support/external_versions.py

```diff
@@ -252,6 +255,43 @@ def keys(self):
     def __contains__(self, item):
         return bool(self[item])
 
+    def add(self, name, func=None, interesting=True):
+        """Add a version checker
+
+        Could be used e.g. by extensions to add important to them custom checks.
+        If will not add `name` if already exists.  If it exists and `func` is
+        different - it will override with a new `func`.
+
+        Parameters
+        ----------
+        name: str
+          Name of the check (usually a name of the Python module, or an
+          external command prefixed with "cmd:")
+        func: callable, optional
+          Function to be called to obtain version information, if it is not
+          a Python module following standard ways to expose version via
+          `__version__`
+        interesting: bool, optional
+          Makes sense only for entries without func - either it is considered
+          "interesting" and should be included in output of dumps
+           (with query=True).
+        """
+        if func:
+            func_existing = self.CUSTOM.get(name, None)
+            was_known = False
+            if func_existing and func_existing is not func:
```

Comment:
Is there much value in supporting overrides?  What about simply raising an exception on a name conflict?

Reply:
what if definition for the same added by two  different extensions/modules but with different implementations?  Exception would cause the entire import to fail, due to possibly unrelated 3rd party extension.  So I thought that better to let it through here

Reply:
All right.